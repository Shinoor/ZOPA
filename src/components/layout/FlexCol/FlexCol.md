### Summary

`<FlexCol />` is meant to be used a child component of `<FlexRow />` ... its padding is offset by `<FlexRow />` _negative_ horizontal margin.

### Examples

- Standard layout

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer gutter={0}>
  <FlexRow>
    <FlexCol xs={12} m={6}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-6</div>
    </FlexCol>
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-3</div>
    </FlexCol>
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-3</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Content based layout ( _collapsing on mobile_ )

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow>
    <FlexCol xs={12} m="fill">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-fill</div>
    </FlexCol>
    <FlexCol xs={12} m="auto">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-auto</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Responsive layout

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow justify="space-between">
    <FlexCol xs={12} m="auto">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-auto</div>
    </FlexCol>
    <FlexCol xs={12} m="auto">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-auto</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Layout with a column hidden on small screens

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow justify="space-between">
    <FlexCol xs="hidden" m={6}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-hidden m-auto</div>
    </FlexCol>
    <FlexCol xs={12} m={6}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-auto</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Layout with column order swapped on small screens

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow direction="row-reverse">
    <FlexCol xs={12} m={6}>
      <div style={{ backgroundColor: '#73E1D6' }}>second column</div>
    </FlexCol>
    <FlexCol xs={12} m={6}>
      <div style={{ backgroundColor: '#73E1D6' }}>first column</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Layout aligned vertically

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow align="center">
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-9</div>
    </FlexCol>
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6', height: '50px' }}>xs-12 m-3</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Each column with different vertical alignment

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow>
    <FlexCol xs={12} m={3} align="flex-end">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-9</div>
    </FlexCol>
    <FlexCol xs={12} m={3} align="center">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-9</div>
    </FlexCol>
    <FlexCol xs={12} m={3} align="flex-start">
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-9</div>
    </FlexCol>
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6', height: '50px' }}>xs-12 m-3</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Layout with custom gutter ( `48px` )

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow gutter={48}>
    <FlexCol xs={12} m={6}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-6</div>
    </FlexCol>
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-3</div>
    </FlexCol>
    <FlexCol xs={12} m={3}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-12 m-3</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```

- Layout with custom number of columns ( `5` )

```tsx
import { FlexContainer, FlexRow, FlexCol } from '@zopauk/react-components';

<FlexContainer>
  <FlexRow cols={5}>
    <FlexCol xs={1}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-1</div>
    </FlexCol>
    <FlexCol xs={1}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-1</div>
    </FlexCol>
    <FlexCol xs={1}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-1</div>
    </FlexCol>
    <FlexCol xs={1}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-1</div>
    </FlexCol>
    <FlexCol xs={1}>
      <div style={{ backgroundColor: '#73E1D6' }}>xs-1</div>
    </FlexCol>
  </FlexRow>
</FlexContainer>;
```
