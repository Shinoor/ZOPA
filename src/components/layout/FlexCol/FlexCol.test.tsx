import React from 'react';
import { render } from '@testing-library/react';
import FlexCol from './FlexCol';

describe('<FlexCol />', () => {
  it('renders without issues', () => {
    const { container } = render(<FlexCol />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with a custom gutter', () => {
    const customGutter = 12;
    const { container } = render(<FlexCol gutter={12}>A column</FlexCol>);

    expect(container.firstChild).toHaveStyleRule('padding-right', `${customGutter}px`);
    expect(container.firstChild).toHaveStyleRule('padding-left', `${customGutter}px`);
  });

  it('renders with custom alignment', () => {
    const customAlignment = 'flex-start';
    const { container } = render(<FlexCol align={customAlignment}>A column</FlexCol>);

    expect(container.firstChild).toHaveStyleRule('align-self', customAlignment);
  });

  it('renders with custom columns', () => {
    const { container } = render(
      <FlexCol cols={4} m={2}>
        A column
      </FlexCol>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders hidden at certain breakpoint', () => {
    const { container } = render(<FlexCol m="hidden">A column</FlexCol>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders filling the columns at certain breakpoint', () => {
    const { container } = render(<FlexCol m="fill">A column</FlexCol>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
