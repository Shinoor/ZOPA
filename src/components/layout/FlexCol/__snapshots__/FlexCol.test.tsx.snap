// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexCol /> renders filling the columns at certain breakpoint 1`] = `
.c0 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

@media (min-width:768px) {
  .c0 {
    display: block;
    -webkit-flex-basis: 0;
    -ms-flex-preferred-size: 0;
    flex-basis: 0;
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    max-width: 100%;
  }
}

<div
  class="c0"
>
  A column
</div>
`;

exports[`<FlexCol /> renders hidden at certain breakpoint 1`] = `
.c0 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

<div
  class="c0"
>
  A column
</div>
`;

exports[`<FlexCol /> renders with custom columns 1`] = `
.c0 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

@media (min-width:768px) {
  .c0 {
    display: block;
    -webkit-flex: 0 0 50%;
    -ms-flex: 0 0 50%;
    flex: 0 0 50%;
    max-width: 50%;
  }
}

<div
  class="c0"
  cols="4"
>
  A column
</div>
`;

exports[`<FlexCol /> renders without issues 1`] = `
.c0 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: px;
  padding-left: px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

<div
  class="c0"
/>
`;
